book/content/binderhub/binderhub.md

```diff
@@ -0,0 +1,159 @@
+
+## How does a BinderHub work?
+
+BinderHub relies on different tools and resources in order to create and launch the Binder instances.
+
+A high-level explanation of the BinderHub architecture can be found [here](https://binderhub.readthedocs.io/en/latest/overview.html).
+
+| ![cloud_neutral_binderhub](/figures/cloud_neutral_binderhub.png) |
+|:---:|
+| A representation of the BinderHub architecture. |
+
+### Compute Resources
+
+BinderHub is cloud-neutral which means it can be deployed on any cloud platform.
+Therefore, the minimum requirement is a subscription to a cloud platform of your choosing.
+
+In fact, BinderHub is not dependent on cloud-hosting at all and can be deployed onto an on-premise computing system.
+
+### Kubernetes
+
+[Kubernetes](https://kubernetes.io/) is a system for automating deployment, scaling (making more or fewer copies), and management of containers across a compute cluster (it doesn't have to be cloud-based).
+BinderHub uses Kubernetes to manage the resources requested by the users of the Binder service, and to support the tools of that build the instances.
+
+### Helm
+
+[Helm](https://helm.sh/) is a package manager for Kubernetes.
+Packages come in the form of Charts which are a set of instructions to deploy, upgrade and manage applications running on a Kubernetes cluster.
+They can make installing and managing Kubernetes applications much easier and Specific Charts for projects can be published online.
+For example, the Helm Chart for BinderHub are available [here](https://jupyterhub.github.io/helm-chart/#development-releases-binderhub).
+
+### JupyterHub
+
+[JupyterHub](https://jupyter.org/hub) is a multi-user server for Jupyter Notebooks and containers alike.
+The JupyterHub's main role is to connect the user's browser to the BinderHub instance running on the Kubernetes cluster.
+However, the JupyterHub can be further customised to provide greater control over the operation of the BinderHub.
+
+### repo2docker
+
+[repo2docker](https://repo2docker.readthedocs.io/en/latest/?badge=latest) is a tool that automatically builds a Docker image from a code repository given a configuration file.
+This Docker container will contain all of the code, data and resources that are listed in the repository.
+All the software required to run the code will also be preinstalled from the configuration file.
+
+### What happens when a Binder link is clicked?
+
+1. The link to the repository is resolved by BinderHub.
+2. BinderHub searches for a Docker image relating to the provided reference (for example, git commit hash, branch or tag).
+3. **If a Docker image is not found**, BinderHub requests resources from the Kubernetes cluster to run repo2docker to do the following:
+   * Fetch the repository,
+   * Build a Docker image containing the software requested in the configuration file,
+   * Push that image to the Docker registry.
```

Comment:
should you define "Docker registry" somewhere?

Reply:
Definition added to glossary table in [`e73d861`](https://github.com/alan-turing-institute/the-turing-way/pull/557/commits/e73d861e400c109dcd995ddd39fca2a279db610c).